app/actors/hyrax/actors/file_actor.rb

```diff
@@ -5,62 +5,66 @@ class FileActor
+      # @example for File
+      #   file_actor = Hyrax::Actors::FileActor.new(fileset, :original_file, user)
+      #   file = File.open('/tmp/mydir/upload.jpg', 'rb')
+      #   io = Hydra::Derivatives::IoDecorator.new(file, 'image/jpeg', 'spirostomum_2400x1200.jpg')
+      #   file_actor.ingest_file(io)
+      # @example for Tempfile
+      #   tempfile = get_tempfile_from_request # presume you get a Tempfile from Rack, Rails or wherever
+      #   io = Hydra::Derivatives::IoDecorator.new(tempfile, 'image/jpeg', File.basename(tempfile.path))
+      #   file_actor.ingest_file(io)
+      # @example for ActionDispatch::Http::UploadedFile
+      #   file = ActionDispatch::Http::UploadedFile.new(filename: '汉字.jpg', type: 'image/jpeg', tempfile: tempfile)
+      #   io = Hydra::Derivatives::IoDecorator.new(file, file.content_type, file.original_filename)
+      #   file_actor.ingest_file(io)
+      # @note the primary requirement of the file passed to Hydra::Derivatives::IoDecorator is to support #read
+      # @todo create a job to monitor the temp directory (or in a multi-worker system, directories!) to prune old files that have made it into the repo
+      def ingest_file(io)
+        # Skip versioning because versions will be minted by VersionCommitter as necessary during save_characterize_and_record_committer.
+        Hydra::Works::AddFileToFileSet.call(file_set,
+                                            io,
+                                            relation,
+                                            versioning: false)
+        return false unless file_set.save
+        repository_file = related_file
+        Hyrax::VersioningService.create(repository_file, user)
+        CharacterizeJob.perform_later(file_set, repository_file.id, path_for(io)) # path hint in case next worker is on same box
         true
       end
 
+      # Reverts file and spawns a job to characterize and create derivatives.
+      # @param [String] revision_id
+      # @return [Boolean] true on success (note: does NOT mean spawned job completed), false otherwise
       def revert_to(revision_id)
-        repository_file = file_set.send(relation.to_sym)
+        repository_file = related_file
         repository_file.restore_version(revision_id)
-
         return false unless file_set.save
-
         Hyrax::VersioningService.create(repository_file, user)
-
-        # Characterize the original file from the repository
         CharacterizeJob.perform_later(file_set, repository_file.id)
         true
       end
 
-      private
-
-        def working_file(file)
-          path = file.path
-          return path if File.exist?(path)
-          Hyrax::WorkingDirectory.copy_file_to_working_directory(file, file_set.id)
-        end
+      # @return [Hydra::PCDM::File] the file referenced by relation
+      def related_file
+        file_set.send(relation) || raise("No .#{relation} returned for FileSet #{file_set.id}")
```

Comment:
This should probably use `public_send`. Is the period in the message intentional?